src/monitor.rs

```diff
@@ -426,6 +440,14 @@ async fn submit_and_watch_solution<T: MinerConfig + Send + Sync + 'static>(
 		.tx()
 		.create_signed_with_nonce(&tx, &*signer, nonce, ExtrinsicParams::default())?;
 
+	let outcome = api.rpc().dry_run(xt.encoded(), None).await?;
```

Comment:
Should be optional, via a flag, in case we are running against a node that does not support unsafe RPCs. 

Perhaps the default can be with dry-run, and `--no-dry-run` flag added.

Or, if you prefer to keep it flexible, we always do-run iff it is available

Lastly, needs to inform devops that our nodes should expose unsafe RPCs :)

Reply:
let's add a flag then forgot that that RPC is unsafe